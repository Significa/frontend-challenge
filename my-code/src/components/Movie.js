import React from 'react'
import { Link} from "react-router-dom";

export function Movie(props) {
    return (
        <tr>
            <td>{props.item.Title}</td>
            <td>{props.item.Year}</td>
            <td>{props.item.Type}</td>
            <Link to={"/title/" + props.item.imdbID}>
                <td><img className="zoom" width="45%" src={props.item.Poster} alt={props.item.Title} title={props.item.Title}></img></td>
            </Link>

        </tr>

    );

}