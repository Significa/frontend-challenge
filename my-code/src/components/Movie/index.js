import React from 'react'
import PropTypes from 'prop-types'

import MovieContent from '../../containers/MovieContent'
import Message from '../Message'

export default function Movie({ loaded, error, errorMessage }) {
  if (error) {
    return <Message title="Something went wrong" subtitle={errorMessage} />
  }

  if (!loaded) return null

  return <MovieContent />
}

Movie.propTypes = {
  loaded: PropTypes.bool.isRequired,
  error: PropTypes.bool.isRequired,
  errorMessage: PropTypes.string
}

Movie.defaultProps = {
  errorMessage: undefined
}
