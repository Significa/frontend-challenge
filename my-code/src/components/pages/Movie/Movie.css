@import '../../../style-vars.css';

.ErrorPage {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  font-size: var(--font-size-4);
  color: var(--color-white);
  height: 100%;
}

.ErrorImg {
  margin-bottom: var(--spacing-5);
  height: 300px;
}

.Movie {
  display: flex;
  flex-direction: column;
  height: 100%;

  @media (--mobile-large) {
    flex-direction: row;
  }
}

.MovieContent {
  flex: 1;
  order: 2;

  @media (--mobile-large) {
    min-width: 400px;
    max-width: 700px;
    order: 1;
    margin: var(--spacing-3) 0;
    color: var(--color-white);
  }
}

.Title {
  font-size: var(--font-size-4);
  margin: 0;
  margin-bottom: var(--spacing-4);
  font-weight: var(--font-weigth-bold);
  color: var(--color-white);

  @media (--tablet) {
    font-size: var(--font-size-5);
  }

  @media (--desktop) {
    font-size: var(--font-size-6);
  }
}

.BackButton {
  background: none;
  padding: 0;
  border: none;
  appearance: none;
  outline: none;

  cursor: pointer;
  margin-bottom: var(--spacing-5);
  height: 24px;
  width: 30px;
}

.MovieInfo {
  display: inline-flex;
  align-items: center;
  margin-bottom: var(--spacing-5);
  color: var(--color-light-grey);
  font-size: var(--font-size-2);
}

.Rated {
  font-size: var(--font-size-1);
  font-weight: var(--font-weigth-medium);
  color: var(--color-dark);
  background-color: var(--color-light-grey);
  padding: var(--spacing-1) var(--spacing-2);
  margin: 0 var(--spacing-3);
  border-radius: var(--border-radius-1);
}

.Badges {
  display: flex;
  flex-wrap: wrap;
  margin-bottom: var(--spacing-6);
}

.Badge,
.LikeButton {
  margin: var(--spacing-2) 0;
  margin-right: var(--spacing-5);
  display: flex;
  align-items: center;
  overflow: hidden;
  border-radius: var(--border-radius-1);
  font-weight: var(--font-weigth-medium);
  border: 1px solid var(--color-mid-grey);
  color: var(--color-white);
}

.LikeButton {
  font-weight: var(--font-weigth-bold);
  padding: var(--spacing-3) var(--spacing-4);
  color: var(--color-light-grey);
  background: none;
  appearance: none;
  outline: none;

  &.liked {
    background: var(--color-red);
    color: var(--color-white);
  }
}

.Ratting {
  padding: var(--spacing-3);
}

.Imdblogo,
.Rottenlogo {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100%;
}

.HeartIcon {
  height: 1.2em;
  padding-right: var(--spacing-3);
}

.Rottenlogo {
  padding: 0 var(--spacing-4);
  background-color: var(--color-red);
}

.Imdblogo {
  padding: 0 var(--spacing-3);
  background-color: var(--color-yellow);
}

.Poster {
  box-sizing: border-box;
  margin: var(--spacing-5) auto;
  max-width: 200px;
  flex: 1;
  order: 1;

  @media (--mobile-large) {
    max-width: 400px;
    margin: var(--spacing-5);
  }
}

.PosterImg {
  box-sizing: border-box;
  height: auto;
  width: 100%;
  object-fit: cover;
  overflow: hidden;
  border-radius: var(--border-radius-2);
}

.PosterImgError {
  border-radius: 5px;
  border: 2px solid var(--color-mid-grey);
}

.MovieDetails {
  max-width: 500px;
}

.Lists {
  display: flex;
  flex-wrap: wrap;
}

.DetailsList {
  color: var(--color-white);
  list-style: none;
  margin-right: 40px;
  padding: 0;

  & li {
    margin: var(--spacing-2) 0;
  }
}

.DetailsHeading,
.PlotHeading {
  color: var(--color-light-grey);
  font-size: var(--font-size-1);
  font-weight: var(--font-weigth-medium);
}

.PlotText {
  color: var(--color-white);
}
