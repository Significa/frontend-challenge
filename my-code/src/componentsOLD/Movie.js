import React,{useEffect,useState} from 'react'
import { useAPIFetch } from "../hooks/useAPIFetch";



export function Movie(props){
    let apiKey = "798e7a67";

    //const [query, setQuery] = useState("Avatar");
    //const [disabled, setDisabled] = useState(true);

    let url = `https://www.omdbapi.com/?apikey=${apiKey}&i=${props.title}`;

    const [data, loading, found] = useAPIFetch(url, props.title);
    return (
        <div>
        <p>{loading && "Loading..."}</p>
        <p>{found && data.Title}</p>
    </div>
    );
}

