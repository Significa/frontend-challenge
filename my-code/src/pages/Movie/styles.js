import styled from 'styled-components';
import variables from '../../styles/variables';

export const Container = styled.div`
  .btn-back {
    border: 0;
    padding-left: 0;
  }

  .movie-poster {
    text-align: center;

    img {
      border-radius: ${variables.BORDER_RADIUS};
      max-width: 100%;
    }
  }

  .movie-attributes {
    color: ${variables.COLOR_LIGHTGREY};

    .movie-attributes-separator {
      padding: 0 6px;
    }

    .movie-rated {
      background-color: ${variables.COLOR_LIGHTGREY}
      border-radius: ${variables.BORDER_RADIUS}
      color: ${variables.COLOR_DARK};
      padding: 4px 8px;
    }
  }

  .movie-title {
    font-size: ${variables.FONT_SIZE_GIANT};
    line-height: ${variables.FONT_SIZE_GIANT};
  }

  .movie-ratings {
    display: flex;
    margin-top: 30px;

    .heart-button {
      border-style: solid;
      border-width: 1px;
      border-color: ${variables.COLOR_MIDGREY};
      padding: 6px 8px;
    }
  }

  .movie-plot {
    margin-top: 30px;
  }

  .movie-people {
    margin-top: 30px;
  }

  span.label {
    color: ${variables.COLOR_LIGHTGREY};
    display: block;
    margin-bottom: 10px;
  }
`;

export const Loading = styled.div`
  text-align: center;
`;
