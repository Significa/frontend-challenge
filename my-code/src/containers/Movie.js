import { connect } from 'react-redux'

import { getIsLoaded } from '../state/modules/movies'
import { getIsFailed, getErrorMessage } from '../state/modules/api-status'
import getQueryParam from '../utils/get-query-param'

import Movie from '../components/Movie'

const mapStateToProps = ({ router, movies, apiStatus }) => ({
  loaded: getIsLoaded(movies, getQueryParam(router.location.search, 'id')),
  error: getIsFailed(apiStatus),
  errorMessage: getErrorMessage(apiStatus)
})

export default connect(mapStateToProps)(Movie)
