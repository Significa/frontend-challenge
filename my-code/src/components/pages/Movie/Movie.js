// @flow
import React from 'react';
import classNames from 'classnames';
import type { MoviePageProps } from '../../../pages/movie/[id]';
import Router from 'next/router';
import arrowIconGrey from '../../../icons/icon-arrow-grey.svg';
import arrowIconWhite from '../../../icons/icon-arrow-white.svg';
import imdblogo from '../../../icons/logo-imdb.svg';
import rottenTomatoesLogo from '../../../icons/logo-rotten-tomatoes.svg';
import heartGreyIcon from '../../../icons/icon-heart-grey.svg';
import heartFullIcon from '../../../icons/icon-heart-full.svg';
import Layout from '../../shared/Layout';
import useLike from '../../hooks/useLike';
import styles from './Movie.css';

type Props = MoviePageProps;

export default function Movie(props: Props) {
  const { movie, error } = props;

  if (error) {
    return (
      <Layout>
        <div className={styles.ErrorPage}>
          <img className={styles.ErrorImg} src={rottenTomatoesLogo} />
          <span>Something went wrong :/</span>
        </div>
      </Layout>
    );
  }

  if (Object.keys(movie).length === 0 && movie?.constructor === Object) {
    return (
      <Layout>
        <div className={styles.ErrorPage}>
          <img className={styles.ErrorImg} src={rottenTomatoesLogo} />
          <span>Movie not found :/</span>
        </div>
      </Layout>
    );
  }

  const { title, runtime, year, id, rated, poster, ratings, plot, actors, genre, director } = movie;
  const actorsList = actors.split(', ');
  const genreList = genre.split(', ');
  const directorList = director.split(', ');

  const { liked, toggle } = useLike(id);

  const handleLike = () => {
    toggle();
  };

  const handleBackHistory = () => Router.back();

  return (
    <Layout>
      <div>
        <button title="Back" className={styles.BackButton} onClick={handleBackHistory}>
          <img
            onMouseOver={(e) => (e.currentTarget.src = arrowIconWhite)}
            onMouseLeave={(e) => (e.currentTarget.src = arrowIconGrey)}
            src={arrowIconGrey}
            alt="Back icon"
            className={styles.ArrowIcon}
          />
        </button>
      </div>
      <div className={styles.Movie}>
        <div className={styles.MovieContent}>
          <div className={styles.MovieInfo}>
            <span>
              {runtime} &#183; {year} &#183;
            </span>
            <span className={styles.Rated}>{rated}</span>
          </div>
          <h1 className={styles.Title}>{title}</h1>
          <section className={styles.Badges}>
            {ratings.length > 0 &&
              ratings.map((rating) => {
                if (rating?.Source === 'Internet Movie Database') {
                  return (
                    <div key={rating.Source} className={styles.Badge}>
                      <div className={styles.Imdblogo}>
                        <img alt="IMDB logo" src={imdblogo} />
                      </div>
                      <span className={styles.Ratting}>{rating?.Value}</span>
                    </div>
                  );
                }

                if (rating?.Source === 'Rotten Tomatoes') {
                  return (
                    <div key={rating?.Source} className={styles.Badge}>
                      <div className={styles.Rottenlogo}>
                        <img atl="Rotten Tomatoes logo" src={rottenTomatoesLogo} />
                      </div>
                      <span className={styles.Ratting}>{rating?.Value}</span>
                    </div>
                  );
                }
              })}
            <button
              title={liked ? 'Added' : 'Add to favourites'}
              onClick={handleLike}
              className={classNames(styles.LikeButton, { [styles.liked]: liked })}
            >
              <img alt="heart icon" className={styles.HeartIcon} src={liked ? heartFullIcon : heartGreyIcon} />
              {liked ? 'Added' : 'Add to favourites'}
            </button>
          </section>
          <section className={styles.MovieDetails}>
            <div>
              <h2 className={styles.PlotHeading}>Plot</h2>
              <p className={styles.PlotText}>{plot}</p>
            </div>
            <div className={styles.Lists}>
              <div>
                <ul className={styles.DetailsList}>
                  <h2 className={styles.DetailsHeading}>Cast</h2>
                  {actorsList.map((cast) => (
                    <li key={cast}>{cast.trim()}</li>
                  ))}
                </ul>
              </div>
              <div>
                <ul className={styles.DetailsList}>
                  <h2 className={styles.DetailsHeading}>Genre</h2>
                  {genreList.map((cast) => (
                    <li key={cast}>{cast.trim()}</li>
                  ))}
                </ul>
              </div>
              <div>
                <ul className={styles.DetailsList}>
                  <h2 className={styles.DetailsHeading}>Director</h2>
                  {directorList.map((cast) => (
                    <li key={cast}>{cast.trim()}</li>
                  ))}
                </ul>
              </div>
            </div>
          </section>
        </div>
        {/** Hack to deal with onError for ssr page */}
        <div
          className={styles.Poster}
          dangerouslySetInnerHTML={{
            __html: `
          <img alt="${title} Poster" src="${poster}"
          class="${styles.PosterImg}"
          onerror="this.parentElement.remove()"
          />
            `
          }}
        />
      </div>
    </Layout>
  );
}
