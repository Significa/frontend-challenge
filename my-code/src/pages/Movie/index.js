import React, { useState, useEffect } from 'react';
import queryString from 'query-string';
import ImageFadeIn from 'react-image-fade-in';

import api, { API_KEY } from '../../services/api';
import history from '../../services/history';
import { Container, Loading } from './styles';
import arrowBack from '../../assets/icon-arrow-grey.svg';
import LikeButton from '../../components/LikeButton';
import MovieRating from '../../components/MovieRating';

export default function Movie(props) {
  // states
  const [isLoading, setIsLoading] = useState(true);
  const [notFound, setNotFound] = useState(false);
  const [movie, setMovie] = useState({});

  useEffect(() => {
    const fetchData = async () => {
      const params = queryString.parse(props.location.search);
      const response = await api.get('/', {
        params: { apikey: API_KEY, i: params.imdbID },
      });

      if (response.data.Response === 'False') {
        setNotFound(true);
      } else {
        setNotFound(false);
        setMovie(response.data);
      }

      setIsLoading(false);
    };

    fetchData();
  }, []);

  /**
   * Handles the click on the back button
   * and send the user back to the results page.
   */
  const handleBack = () => {
    history.push('/');
  };

  return (
    <Container>
      <div>
        <button className="btn-back" type="button" onClick={() => handleBack()}>
          <img src={arrowBack} alt="Back button" />
        </button>
      </div>

      {notFound && <div className="not-found">Movie not found.</div>}

      {isLoading && <Loading>Loading movie information...</Loading>}

      {!isLoading && !notFound && (
        <div className="p-grid">
          <div className="p-col-12 p-col-order-2 p-sm-order-1 p-md-7 p-lg-8">
            <div className="movie-attributes">
              <span className="movie-runtime">{movie.Runtime}</span>
              <span className="movie-attributes-separator">&middot;</span>
              <span className="movie-year">{movie.Year}</span>
              <span className="movie-attributes-separator">&middot;</span>
              <span className="movie-rated">{movie.Rated}</span>
            </div>
            <div className="movie-title">{movie.Title}</div>
            <div className="movie-ratings">
              <div className="p-grid p-nogutter">
                {movie.Ratings.map(rating => {
                  if (rating.Source !== 'Metacritic') {
                    return (
                      <div className="p-col" key={rating.Source}>
                        <MovieRating rating={rating} />
                      </div>
                    );
                  }
                  return '';
                })}

                <div className="p-col p-col-12 p-lg-5 p-xl-7">
                  <LikeButton imdbID={movie.imdbID} showText alwaysVisible />
                </div>
              </div>
            </div>
            <div className="movie-plot">
              <span className="label">Plot</span>
              {movie.Plot}
            </div>
            <div className="movie-people">
              <div className="p-grid">
                <div className="p-col-12 p-md-5">
                  <span className="label">Cast</span>
                  {movie.Actors.split(',').map(item => {
                    return <p key={item}>{item}</p>;
                  })}
                </div>
                <div className="p-col-12 p-md-3">
                  <span className="label">Genre</span>
                  {movie.Genre.split(',').map(item => {
                    return <p key={item}>{item}</p>;
                  })}
                </div>
                <div className="p-col-12 p-md-4">
                  <span className="label">Director</span>
                  {movie.Director.split(',').map(item => {
                    return <p key={item}>{item}</p>;
                  })}
                </div>
              </div>
            </div>
          </div>
          <div className="p-col-12 p-col-order-1 p-sm-order-2 p-md-5 p-lg-4">
            <div className="movie-poster">
              {movie.Poster !== 'N/A' && (
                <ImageFadeIn
                  src={movie.Poster}
                  opacityTransition={1}
                  alt="Poster"
                />
              )}

              {movie.Poster === 'N/A' && (
                <div className="movie-no-poster">(No poster)</div>
              )}
            </div>
          </div>
        </div>
      )}
    </Container>
  );
}
