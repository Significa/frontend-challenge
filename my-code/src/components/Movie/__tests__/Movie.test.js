import React from 'react'
import { shallow } from 'enzyme'

import Movie from '../'

describe('Movie', () => {
  it('renders correctly', () => {
    const tree = shallow(<Movie loaded error={false} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders not loaded correctly', () => {
    const tree = shallow(<Movie loaded={false} error={false} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders error correctly', () => {
    const tree = shallow(
      <Movie loaded={false} error errorMessage="Error message goes here" />
    )
    expect(tree).toMatchSnapshot()
  })
})
