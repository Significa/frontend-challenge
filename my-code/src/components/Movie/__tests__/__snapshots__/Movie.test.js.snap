// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie renders correctly 1`] = `<Connect(MovieContent) />`;

exports[`Movie renders error correctly 1`] = `
<Message
  subtitle="Error message goes here"
  title="Something went wrong"
/>
`;

exports[`Movie renders not loaded correctly 1`] = `""`;
